google/cloud/_http.py

```diff
@@ -211,8 +211,12 @@ def build_api_url(
         )
 
         query_params = query_params or {}
-        if query_params:
-            url += "?" + urlencode(query_params, doseq=True)
+
+        if "prettyPrint" not in query_params:
```

Comment:
Trying to think of cases where this could alter requests we're not anticipating and may have issue with the prettyprint param.

* will this interfere with media uploads?  Mainly thinking about the resumable upload scenario.
* will this interfere with oauth negotiation?

Reply:
We could make a pre-release and verify that the tests for other libs that depend on `google-cloud-core` still pass. (The pre-release can also be given to the customer to pin explicitly).

Reply:
> will this interfere with media uploads? Mainly thinking about the resumable upload scenario.

Since [google-resumable-media](https://github.com/googleapis/google-resumable-media-python) uses "requests" directly, this shouldn't affect it.

I've tested these changes locally with the BigQuery system & sample tests, which all pass and do several load jobs.

I'm trying to test locally with the Storage system tests, but seem to be missing some setup steps around their encryption-related tests.

> will this interfere with oauth negotiation?

No, since [google-auth](https://github.com/googleapis/google-auth-library-python) uses "requests" directly. It doesn't go through this library. Even if it did, I believe they've actually moved to a One Platform API for fetching tokens now. (Guessing based on the new-ish oauth2.googleapis.com endpoint)

Reply:
Created a pre-release at https://github.com/googleapis/python-cloud-core/pull/31